scripts/cluster-upgrade.sh

```diff
@@ -310,18 +310,24 @@ upgrade() {
     log $OCM_NAME "upgrade" "ClusterVersion on $TO"
     log $OCM_NAME "upgrade" "checking node versions"
 
-    # fun fact!  after clusterversion says it is done, individual nodes could still be updated.
-    MACHINE_COUNT_ALL=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get nodes --no-headers | wc -l)
+    MASTER_MACHINEPOOL_TOTAL_COUNT=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get machineconfigpools -o json  | jq -r ".items[] | select(.metadata.name == \"master\" ) | .status.machineCount")
+    MASTER_MACHINEPOOL_UPDATED_COUNT=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get machineconfigpools -o json  | jq -r ".items[] | select(.metadata.name == \"master\" ) | .status.updatedMachineCount")
 
-    MACHINE_COUNT_UPGRADED=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get nodes -o json | jq -r ".items[].status | select(.annotations[\"machineconfiguration.openshift.io/currentConfig\"] == .annotations[\"machineconfiguration.openshift.io/desiredConfig\"]) | .conditions[] | select(.type == \"Ready\" and .status == \"True\") | .reason" | wc -l)
+    while [ $MASTER_MACHINEPOOL_UPDATED_COUNT -lt $MASTER_MACHINEPOOL_TOTAL_COUNT ];
+    do
+        # wait for all masters to be upgraded
+        log $OCM_NAME "upgrade" "master nodes upgraded: $MASTER_MACHINEPOOL_UPDATED_COUNT/$MASTER_MACHINEPOOL_TOTAL_COUNT"
+        sleep 60
+    done
 
-    MACHINE_COUNT_PENDING=$((MACHINE_COUNT_ALL-MACHINE_COUNT_UPGRADED))
+    WORKER_MACHINEPOOL_TOTAL_COUNT=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get machineconfigpools -o json  | jq -r ".items[] | select(.metadata.name == \"worker\" ) | .status.machineCount")
+    WORKER_MACHINEPOOL_UPDATED_COUNT=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get machineconfigpools -o json  | jq -r ".items[] | select(.metadata.name == \"worker\" ) | .status.updatedMachineCount")
```

Comment:
This is similar enough to the code on L313-4 that the helper function suggested above could take a parameter ("worker" or "master") and DRY even more.

Reply:
Not going to spend time to refactor as it's "good enough" and our end-goal is to delete this.. we don't need it to be as optimized as we can get it, just need it to work and be a reference for the upgrade operator we're designing.